code/game/objects/items/food/donkpocket.dm

```diff
@@ -22,6 +22,32 @@
 	microwaved_type = null
 	tastes = list("meat" = 2, "dough" = 2, "laziness" = 1)
 	foodtypes = GRAIN
+	///Type of donkpocket that we will turn back into when we cool down!
+	var/donkpocket_basetype = /obj/item/food/donkpocket
+	///Tracks how long till we cool down!
+	var/cooldown_cooldown
+
+/obj/item/food/donkpocket/warm/examine(mob/user)
+	. = ..()
+	var/time_left = COOLDOWN_TIMELEFT(src, cooldown_cooldown)
+	if(time_left < 1.5 MINUTES)
+		. += "[src] is getting cold!"
+	else if(time_left < 3 MINUTES)
+		. += "[src] is still warm!"
+	else if(time_left < 5 MINUTES)
+		. += "[src] is quite hot!"
+	else
+		. += "[src] is steaming hot!"
+
+/obj/item/food/donkpocket/warm/Initialize(mapload)
+	. = ..()
+	COOLDOWN_START(src, cooldown_cooldown, 7 MINUTES)
+	addtimer(CALLBACK(src, .proc/cooldown_and_revert_to_basetype), 7 MINUTES)
+
+///Changes the item back to its not warm state.
+/obj/item/food/donkpocket/warm/proc/cooldown_and_revert_to_basetype()
+	new donkpocket_basetype(loc)
```

Comment:
I've got a bad feeling about this. What happens if it's in a container, like a box or a backpack? What if it's a variable on some object (like "current target" or whatever) and gets replaced? Donk pockets are also multiple bites, so this'll regen it.

The proper way to do this IMO is to have this proc just change the reagents, and maybe the description (already done)/name as well.

Reply:
Second thought lets just nuke it.

Modifying the reagents would be simple enough, but you can inject chemicals into food and all that's probably gonna be a pain to track with what's been consumed.